### ISSUE TITLE
PyAV 7.x / dropping Python 2

### ISSUE THREAD

**Issue by jlaine** [mikeboers/PyAV#504]
Our build matrix is getting pretty complex with the support of multiple Python and ffmpeg versions. With Python 2 approaching end of life, I'd like to suggest the we focus our efforts on Python 3 only going forward. PyAV 6.x is mature and if some critical bug crops up we can always issue a patch release to address it.

 I'd also suggest we remove Python 3.4 from our build matrix, stable versions of Linux distros carry Python 3.5+ nowadays.

@mikeboers what are your thoughts about making "develop" Python 3 only?

**Comment by mikeboers** [comment]
My unfortunate contribution is that, professionally, I'm still kinda stuck in 2.7. It's not 100% binding though.

My professional VFX work uses Maya, Nuke, and Houdini, all of which follow the [VFX reference platform](https://www.vfxplatform.com/), which is still on 2.7.

While I can't actually think of anything that I've ever done where I used PyAV in any of those contexts directly, those contexts influence the whole environment. Everything at my last large client is fully solidified into 2.7; their deployment system will only deal with one Python at a time (and even that thing is very likely baked into 2.7).

I'll also lay out that the core of the feature development for PyAV was driven by requirements in my VFX work. So while v6.x would continue to work, much of the pressure I might feel to add new features would be in conflict with the fact that I can't do it on `develop`.

I don't currently have any such work, because I'm trying to finish up [my own project](http://shadowsinthegrass.com/), but the environment for that project on my local systems and my farm is based on the same tech.

I might also not go back to VFX work like that. I'm currently dreaming of blue skies and new types of work once I'm done my project.

Despite that huge influence on my development life, all of my new non-VFX projects are (finally) Python 3. I'm nearly a decade late on that, but it is wonderful to finally have done it.

In summary: my immediate gut reaction is to ask that we leave a single build in Travis with 2.7. Asking that I can see that it might be a lot though, so I could be convinced. I'm just laying all of my cards out for a complete discussion.

**Comment by mikeboers** [comment]
Lets also drop FFmpeg 3.2 and 3.3. They started at least 2 years ago.

**Comment by KengoSawa2** [comment]
Hi, Thank you great library contribution.
I'm using PyAV to make my tools.

As for the continued support for Python 2.7,
I think Mike is right.
The VFX Reference Platform requires Python 2.7 within 2019.

https://www.vfxplatform.com/

It will finally support Python 3 in 2020, but I think it's very difficult.
Since the industry's in-house tools, workflow, are too optimized for Python 2.7, 
migration is very difficult.

People in the CG / VFX world are not limited to using PyAV, but I think that the world to which the original author Mike belongs can be respected.

By the way, I am not a CG / VFX expert but a post production IT administrator.

**Comment by jlaine** [comment]
Python 2 support is gone as of PyAV 8.0.0. Stick to the older releases if you are stuck with EOL Python.
